src/layers/s2-geometry-layer/s2-geometry-layer.js

```diff
@@ -0,0 +1,231 @@
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {S2Layer} from 'deck.gl';
+import {hexToRgb} from 'utils/color-utils';
+
+import Layer from '../base-layer';
+import S2LayerIcon from './s2-layer-icon';
+import {HIGHLIGH_COLOR_3D} from 'constants/default-settings';
+
+export const S2_TOKEN_FIELDS = {
+  token: [
+    's2',
+    's2_token'
+  ]
+};
+
+export const s2RequiredColumns = ['token'];
+export const S2TokenAccessor = ({token}) => d => d[token.fieldIdx];
+
+export const S2VisConfigs = {
+  opacity: 'opacity',
+  colorRange: 'colorRange',
+  filled: {
```

Comment:
if we support `filled ` toggle then we should also support `stroked` and add `strokeWidth` support (see geoson layer). Otherwise, when `filled:false`there will be no geometry shown on the screen

Reply:
@macrigiuseppe allow `stroked` and `strokeColor`, `strokeWidth` based on

Reply:
@macrigiuseppe if you are not going to add those, just delete `filled`